src/code/z_jpeg.c

```diff
@@ -90,6 +95,9 @@ void Jpeg_CopyToZbuffer(u16* src, u16* zbuffer, s32 x, s32 y) {
     }
 }
 
+/**
+ * Gets an u16 from an (possibly unaligned) address in memory, avoiding unaligned memory accesses.
+ */
 u16 Jpeg_GetU16(u8* ptr) {
     if (((u32)ptr & 1) == 0) { // if the address is aligned to 2
         return *(u16*)ptr;
```

Comment:
if i had to guess, this probably has a more specific purpose than just "get any u16", and can probably be named accordingly. But I havent read the system at all so I could be wrong

Reply:
I think the only real reason for the existence of this function is for the ability to read an unaligned u16 without crashing the CPU due to use of the `lhu` instruction (as the already existing comment suggests)

Reply:
that sounds better yea. though as you noted in your comment, it may not be unaligned. So that can possibly be confusing.
Imo id say leave what you have for now and someone else who reviews this can give their thoughts as well, Im not sure which is better

Reply:
I've been skimming through the rest of the source code, it seems like there are more situations like this, where an unaligned 16-bit access is needed. As far as I can see, those are all inlined though.

Reply:
>that sounds better yea. though as you noted in your comment, it may not be unaligned. So that can possibly be confusing.

That's true, although I feel the documentation comment clarifies this by saying "possibly unaligned"

Reply:
I'll just change it to GetUnalignedU16 with some slightly improved documentation - I'll see what others think of it :slightly_smiling_face: